frontend/libs/studio-components/src/components/StudioProfileMenu/StudioProfileMenu.test.tsx

```diff
@@ -0,0 +1,235 @@
+    await user.click(triggerButton);
+
+    const dividers = screen.getAllByRole('separator');
+    expect(dividers.length).toBe(1);
+  });
+
+  it('should close the menu when clicking a menuitem', async () => {
+    const user = userEvent.setup();
+    renderStudioProfileMenu();
+
+    expect(screen.queryByRole('menuitem', { name: menuItem1 })).not.toBeInTheDocument();
+
+    const triggerButton = screen.getByRole('button', { name: mockTriggerButtonText });
+    await user.click(triggerButton);
+
+    expect(screen.getByRole('menuitem', { name: menuItem1 })).toBeInTheDocument();
+    await user.click(screen.getByRole('menuitem', { name: menuItem1 }));
+
+    expect(mockOnClick).toHaveBeenCalledTimes(1);
+    expect(screen.queryByRole('menuitem', { name: menuItem1 })).not.toBeInTheDocument();
+  });
+
+  it('should render aria-label on trigger button', () => {
+    renderStudioProfileMenu();
+
+    const triggerButton = screen.getByRole('button', { name: mockTriggerButtonText });
+    expect(triggerButton).toHaveAttribute('aria-label', mockAriaLabel);
+  });
+
+  it('should truncate triggerButtonText if it exceeds "truncateAt" characters', () => {
+    const longText = 'This is a very long trigger button text that exceeds 30 characters';
+    const truncateAt: number = 30;
+
+    renderStudioProfileMenu({ triggerButtonText: longText, truncateAt });
+
+    const truncatedText = `${longText.slice(0, truncateAt)}...`;
+    expect(screen.getByText(truncatedText)).toBeInTheDocument();
+  });
+
+  it('should not truncate triggerButtonText if the text is smaller than truncateAt', () => {
+    const shortText = 'Short text';
+    const truncateAt: number = 30;
+
+    renderStudioProfileMenu({ triggerButtonText: shortText, truncateAt });
+
+    const truncatedText = `${shortText.slice(0, truncateAt)}...`;
+    expect(screen.queryByText(truncatedText)).not.toBeInTheDocument();
+    expect(screen.getByText(shortText)).toBeInTheDocument();
+  });
+
+  it('should toggle the dropdown menu open and close when the trigger button is clicked multiple times', async () => {
+    const user = userEvent.setup();
+    renderStudioProfileMenu();
+
+    const triggerButton = screen.getByRole('button', { name: mockTriggerButtonText });
+
+    await user.click(triggerButton);
+    expect(screen.getByRole('menuitem', { name: menuItem1 })).toBeInTheDocument();
+
+    await user.click(triggerButton);
+    expect(screen.queryByRole('menuitem', { name: menuItem1 })).not.toBeInTheDocument();
+  });
+
+  it('should close the dropdown menu when handleClose is called', async () => {
+    const user = userEvent.setup();
+    renderStudioProfileMenu();
+
+    const triggerButton = screen.getByRole('button', { name: mockTriggerButtonText });
+    await user.click(triggerButton);
+
+    expect(screen.getByRole('menuitem', { name: menuItem1 })).toBeInTheDocument();
+
+    const handleCloseButton = screen.getByRole('button', { name: mockTriggerButtonText });
```

Comment:
Nitpicking, but `triggerButton` can be reused here 🧐

Reply:
Great spotted @ErlingHauan! 🤩